src/test/scala/chiselTests/LFSR16.scala

```diff
@@ -6,43 +6,60 @@ import chisel3._
 import chisel3.testers.BasicTester
 import chisel3.util._
 
-class LFSR16 extends Module {
-  val io = IO(new Bundle {
-    val inc = Input(Bool())
-    val out = Output(UInt(16.W))
-  })
-  val res = RegInit(1.U(16.W))
-  when (io.inc) {
-    val nxt_res = Cat(res(0)^res(2)^res(3)^res(5), res(15,1))
-    res := nxt_res
+/**
+  * This test creates two 4 sided dice.
+  * Each cycle it adds them together and adds a count to the bin corresponding to that value
+  * The asserts check that the bins show the correct distribution.
+  */
+//scalastyle:off magic.number
+class LFSRTester extends BasicTester {
+  val bins = Reg(Vec(8, UInt(32.W)))
+
+  // Use tap points on each LFSR so values are more independent
+  val die0 = Cat(Seq.tabulate(2) { i => LFSR16()(i) })
+  val die1 = Cat(Seq.tabulate(2) { i => LFSR16()(i + 2) })
+
+  val (trial, done) = Counter(true.B, 10000)
+
+  val rollValue = die0 +& die1  // Note +& is critical because sum will need an extra bit.
+
+  bins(rollValue) := bins(rollValue) + 1.U
+
+  when(done) {
+    printf("bins: %d %d %d %d %d %d %d %d\n",
+      bins(0), bins(1), bins(2), bins(3), bins(4), bins(5), bins(6), bins(7))
```

Comment:
I would note that this is more or less what printable is for, you could just do:
```scala
printf(p"bins: $bins\n")
```
I'll do something like:
```
bins: Vec(125, 187, ...)
```